BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-bungeecord</artifactId>
++            <version>4.0.0-SNAPSHOT</version>
++        </dependency>
+     </dependencies>
+ </project>
+diff --git a/api/src/main/java/net/md_5/bungee/api/CommandSender.java b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+index d5888311..a8224a9f 100644
+--- a/api/src/main/java/net/md_5/bungee/api/CommandSender.java
++++ b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+@@ -1,9 +1,13 @@
+ package net.md_5.bungee.api;
+ 
+ import java.util.Collection;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.text.Component;
++// Waterfall end
+ import net.md_5.bungee.api.chat.BaseComponent;
+ 
+-public interface CommandSender
++public interface CommandSender extends Audience // Waterfall - we're going on an adventure!
+ {
+ 
+     /**
+@@ -34,14 +38,18 @@ public interface CommandSender
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link Audience#sendMessage(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent... message);
+ 
+     /**
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link Audience#sendMessage(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent message);
+ 
+     /**
+diff --git a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+index a4011335..f20f401b 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
++++ b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+@@ -4,11 +4,19 @@ import com.google.common.base.Preconditions;
+ import java.io.File;
+ import java.net.InetSocketAddress;
+ import java.net.SocketAddress;
++import java.text.MessageFormat;
+ import java.util.Collection;
++import java.util.Locale;
+ import java.util.Map;
+ import java.util.UUID;
+ import java.util.logging.Logger;
+ import lombok.Getter;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.audience.ForwardingAudience;
++import net.kyori.adventure.text.Component;
++// Waterfall end
++import net.kyori.adventure.text.TranslatableComponent;
+ import net.md_5.bungee.api.chat.BaseComponent;
+ import net.md_5.bungee.api.config.ConfigurationAdapter;
+ import net.md_5.bungee.api.config.ServerInfo;
+@@ -16,8 +24,9 @@ import net.md_5.bungee.api.connection.ProxiedPlayer;
+ import net.md_5.bungee.api.plugin.Plugin;
+ import net.md_5.bungee.api.plugin.PluginManager;
+ import net.md_5.bungee.api.scheduler.TaskScheduler;
++import org.checkerframework.checker.nullness.qual.NonNull;
```

Comment:
I suggest either not importing or putting all new imports at the end of the import list. That will prevent eventual future merge issues. And the waterfall comments are completely out of place.